packages/flutter_tools/lib/src/dart/dependencies.dart

```diff
@@ -7,6 +7,22 @@ import 'package:analyzer/analyzer.dart' as analyzer;
 import '../base/file_system.dart';
 import '../dart/package_map.dart';
 
+// List of flutter specific environment configurations.
+// See https://github.com/munificent/dep-interface-libraries/blob/master/Proposal.md
+final List<String> _configurationConstants = <String>['dart.library.io'];
```

Comment:
Maybe  add a TODO or comment stating that we would have to populate configurationConstants with all the dart: libraries, potentially "dart.IsFlutterEngine" and account for strings in Platform.environment